pulsar-client-cpp/tests/CMakeLists.txt

```diff
@@ -25,4 +25,4 @@ add_executable(main ${TEST_SOURCES})
 
 target_include_directories(main PRIVATE ${CMAKE_SOURCE_DIR}/lib)
 
-target_link_libraries(main ${CLIENT_LIBS} ${GMOCK_LIBRARY_PATH})
+target_link_libraries(main ${CLIENT_LIBS} pulsarShared ${GMOCK_LIBRARY_PATH})
```

Comment:
seems unrelated.

Reply:
Yes, this is slightly unrelated, moving to its own PR.